source/extensions/filters/http/grpc_json_transcoder/json_transcoder_filter.cc

```diff
@@ -203,6 +204,9 @@ ProtobufUtil::Status JsonTranscoderConfig::createTranscoder(
     status = type_helper_->ResolveFieldPath(*request_info.message_type, binding.field_path,
                                             &resolved_binding.field_path);
     if (!status.ok()) {
+      if (ignore_unknown_query_parameters_) {
```

Comment:
This won't work as you expected, take a look on the `ResolveFieldPath` implementation:
https://github.com/grpc-ecosystem/grpc-httpjson-transcoding/blob/master/src/type_helper.cc#L146

It will return when it see the first unknown parameter and then ignore the rest, so this ignores all parameters after the first unknown parameter. You'll have to implement this in the transcoding library (perhaps add `ignore_unknown_query_parameters_` there, and update the dependency in Envoy.

Reply:
Good catch, thanks! 
I was just testing it here.
```
curl "localhost:51051/test?unknown1=asdf&names=bla,blub&unknown2=etcd&unknown3=asdf"
```

This is mapped to a gRPC service with the request:
```
message TestRequest {
  string name = 1;

  string names = 2;
}
```

This worked as expected. How could I test the behavior you mentioned?

Reply:
I tested a bit an here are my findings:

ResolveFieldPath returns once it cannot find any of the field path segments it gets passed.
I was unclear what that actually is but after some fiddling, I found field path can have more than 1 element in this case:

```
message dummy {
string name = 1;
}
message Request {
string id = 1; 
dummy dummy = 2;
}
```
Now if you send a request with `?dummy.name=test&a=b` `field_path` will have 2 elements `[dummy, name]` 
`ResolveFieldPath` will check those in order, if it cannot find `dummy` it will return.

This seems like the behavior is OK. The loop will continue and call `ResolveFieldPath` with `field_path = [a]`.  If there is no field `dummy` on the first level then we don't have to check for `name`. 

What do you think? Maybe I am missing something.

Reply:
Ah sorry I spoke to soon, confused by the prevoius comment of "don't merge yet". This is in loop running over all parameter so I guess it's fine. Can you add that case (known parameter after unknown one) to the test? Thanks.

(btw, please not force-push or delete comment, keep the history please.)

Reply:
Oh sorry, the "don't merge yet" comment I deleted after one minute because I made a mistake on my side, wasn't expecting anyone to have read it.

I will keep that in mind! First force push I did because the CI told me to fix my commit with that (signoff).